i18n/helpers.py

```diff
@@ -31,3 +32,47 @@ def get_language_from_default_language_field(
     else:
         raise ValueError('Invalid default_language for', instance, default_language)
     return default_language
+
+
+def convert_language_code(language_code: str, output_format: Literal['kausal', 'django', 'modeltrans', 'next.js']) -> str:
+    """
+    Convert given language code to wanted output format.
+
+    Args:
+        language_code (str): The language code to convert.
+        output_format (str): The output format to which to convert the language code to. The formats are:
+            'kausal': The format to use internally. (e.g. 'es-US')
+            'django': The format used by Django, e.g. django.utils.translation.get_language. ('es-us')
+            'modeltrans': The format used by django-modeltrans. ('es_us')
+            'next.js': The format used by next.js. ('es-US')
+
+    Returns:
+        Given language code converted to wanted format as a string.
+
+    Raises:
+        ValueError: If language_code or output_format are invalid.
+
+    """
+    regex_match = re.match(r'^([a-zA-Z]{2})(?:[_-]{1}([a-zA-Z]{2}))?$', language_code)
```

Comment:
Probably unnecessary optimization, but you could compile the regexp separately outside the function and store the compiled one in LANGUAGE_CODE_REGEXP or something. But I'm pretty sure this won't matter in practice.

Reply:
Maybe not, Python claims to do some caching for recent patterns as well: https://docs.python.org/3/library/re.html#re.compile

Anyway, good to know as I didn't know there was a potential performance issue with regex patterns. Also I liked the readability of it too so I made the change anyway :smile: